src/main/scala/io/getclump/Tuples.scala

```diff
@@ -49,4 +49,19 @@ protected[getclump] trait Tuples {
   protected[getclump] def fetch4[A, B, C, D, E, F](fetch: (A, B, C, D, E) => Future[Iterable[F]]) = (params: (A, B, C, D), values: E) => (params, values) match {
     case ((a, b, c, d), e) => fetch(a, b, c, d, e)
   }
+
+  protected[getclump] def adapt[K, V](fetch: K => Future[V]) = (keys: Set[K]) =>
+    Future.sequence(keys.toSeq.map(key => fetch(key).map(value => key -> value))).map(_.toMap)
+
+  protected[getclump] def adapt1[A, K, V](fetch: (A, K) => Future[V]) = (a: A, keys: Set[K]) =>
+    Future.sequence(keys.toSeq.map(key => fetch(a, key).map(value => key -> value))).map(_.toMap)
+
+  protected[getclump] def adapt2[A, B, K, V](fetch: (A, B, K) => Future[V]) = (a: A, b: B, keys: Set[K]) =>
+    Future.sequence(keys.toSeq.map(key => fetch(a, b, key).map(value => key -> value))).map(_.toMap)
+
+  protected[getclump] def adapt3[A, B, C, K, V](fetch: (A, B, C, K) => Future[V]) = (a: A, b: B, c: C, keys: Set[K]) =>
+    Future.sequence(keys.toSeq.map(key => fetch(a, b, c, key).map(value => key -> value))).map(_.toMap)
+
+  protected[getclump] def adapt4[A, B, C, D, K, V](fetch: (A, B, C, D, K) => Future[V]) = (a: A, b: B, c: C, d: D, keys: Set[K]) =>
+    Future.sequence(keys.toSeq.map(key => fetch(a, b, c, d, key).map(value => key -> value))).map(_.toMap)
```

Comment:
I was sure I would be able to refactor these different adapt methods for re-use (they are the same apart from the application of the `fetch` function), but I was unable to do so due to the fact that they are returning functions, rather than applying them. If one of you can see a way to make this more DRY, well that would be delightful :)